compiler/src/dotty/tools/dotc/transform/Splicing.scala

```diff
@@ -188,15 +188,19 @@ class Splicing extends MacroTransform:
    *  ```
    */
   private class SpliceTransformer(spliceOwner: Symbol, isCaptured: Symbol => Boolean) extends Transformer:
-    private var refBindingMap = mutable.Map.empty[Symbol, (Tree, Symbol)]
+    private val typeBindingMap = mutable.Map.empty[Symbol, (Tree, Symbol)]
+    private val termBindingMap = mutable.Map.empty[Symbol, (Tree, Symbol)]
     /** Reference to the `Quotes` instance of the current level 1 splice */
     private var quotes: Tree | Null = null // TODO: add to the context
     private var healedTypes: QuoteTypeTags | Null = null // TODO: add to the context
 
     def transformSplice(tree: tpd.Tree, tpe: Type, holeIdx: Int)(using Context): tpd.Tree =
       assert(level == 0)
       val newTree = transform(tree)
-      val (refs, bindings) = refBindingMap.values.toList.unzip
+      val (typeRefs, typeBindings) = typeBindingMap.values.toList.unzip
+      val (termRefs, termBindings) = termBindingMap.values.toList.unzip
```

Comment:
Not new in this PR, but Maps by default don't have a deterministic iteration order. Shouldn't a LinkedHashMap be used for determinism?